bser/tap.go

```diff
@@ -0,0 +1,68 @@
+package bser
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"reflect"
+)
+
+// Tap sends all PDUs sent in either direction along rw
+// through rfn for reads and wfn for writes
+func Tap(rw io.ReadWriter, rfn func([]byte), wfn func([]byte)) io.ReadWriter {
```

Comment:
can the first arg and return value be of type [io.ReadWriteCloser](https://golang.org/pkg/io/#ReadWriteCloser) instead? The `net.Conn` that we pass in already implements this interface